modules/localgov_geo_update/src/MigrateDisplayModes.php

```diff
@@ -0,0 +1,196 @@
+<?php
+
+namespace Drupal\localgov_geo_update;
+
+use Drupal\Core\Entity\Entity\EntityFormDisplay;
+use Drupal\Core\Entity\Entity\EntityFormMode;
+use Drupal\Core\Entity\Entity\EntityViewDisplay;
+use Drupal\Core\Entity\Entity\EntityViewMode;
+
+/**
+ * Migrates display modes for form and view from localgov_geo to geo_entity.
+ */
+class MigrateDisplayModes {
+
+  /**
+   * Migrate localgov_geo display mode to geo_entity.
+   *
+   * Calling this function will overwrite a geo_entity display mode with the
+   * corresponsing display from localgov_geo.
+   *
+   * @param string $bundle
+   *   The localgov_geo bundle name.
+   * @param string $type
+   *   Display mode type (either 'view' or 'form').
+   * @param string $name
+   *   Display mode name (eg. 'default').
+   */
+  public static function migrate(string $bundle, string $type, string $name) {
+
+    // $type must be of type view or form.
+    if ($type != 'view' && $type != 'form') {
+      throw new \Exception('Only form and view display modes can be migrated');
+    }
+
+    // Load the config.
+    $config_manager = \Drupal::service('config.manager');
+    $config_factory = $config_manager->getConfigFactory();
+    $current_config_name = 'core.entity_' . $type . '_display.localgov_geo.' . $bundle . '.' . $name;
+    $current_display_config = $config_factory->get($current_config_name);
+
+    // Load existing display mode to check that it is a valid display mode.
+    if ($type == 'view') {
+      $display = EntityViewDisplay::load('localgov_geo.' . $bundle . '.' . $name);
+    }
+    elseif ($type == 'form') {
+      $display = EntityFormDisplay::load('localgov_geo.' . $bundle . '.' . $name);
+    }
+
+    // Throw error if not a display mode.
+    if (!$display instanceof EntityFormDisplay && !$display instanceof EntityViewDisplay) {
+      throw new \Exception('Provided ' . $type . ' display mode ' . $current_config_name . ' is not defined for localgov_geo bundle ' . $bundle . '.');
+    }
+
+    // Migrate the base display here.
+    static::migrateBaseDisplay($type, $name);
+
+    $new_config_name = 'core.entity_' . $type . '_display.geo_entity.' . $bundle . '.' . $name;
+    $new_display_config = $config_factory->getEditable($new_config_name);
```

Comment:
Where was it that we ended up needing to add UUIDs to entities created this way? I'm guessing we need to do this directly here, I can't remember a better way being bought up.